synapse/replication/http/membership.py

```diff
@@ -88,6 +89,73 @@ async def _handle_request(self, request, room_id, user_id):
         return 200, {"event_id": event_id, "stream_id": stream_id}
 
 
+class ReplicationRemoteKnockRestServlet(ReplicationEndpoint):
+    """Perform a remote knock for the given user on the given room
+
+    Request format:
+
+        POST /_synapse/replication/remote_knock/:room_id/:user_id
+
+        {
+            "requester": ...,
+            "remote_room_hosts": [...],
+            "content": { ... }
+        }
+    """
+
+    NAME = "remote_knock"
+    PATH_ARGS = ("room_id", "user_id")
+
+    def __init__(self, hs):
+        super().__init__(hs)
+
+        self.federation_handler = hs.get_federation_handler()
+        self.store = hs.get_datastore()
+        self.clock = hs.get_clock()
+
+    @staticmethod
+    async def _serialize_payload(  # type: ignore
+        requester: Requester,
+        room_id: str,
+        user_id: str,
```

Comment:
Why pass the `room_id` and `user_id` here if they're unused?

Reply:
I've done so for the knocking classes in 3fb055c92adbf12e5c5a0388c1d7e934b5b89e33, and I've made a PR to do the same for the other classes in this file here: https://github.com/matrix-org/synapse/pull/8809